design/one-pager-resource-connectivity-mvp.md

```diff
@@ -0,0 +1,653 @@
+  exposed in the configuration of their associated resource class.
+
+With the above in place resource connectivity between a `KubernetesCluster` and
+a `MySQLInstance` can be configured roughly as follows:
+
+1. The cloud administrator creates any managed resources necessary to connect
+   a Kubernetes cluster to a MySQL database in their cloud of choice, for
+   example they create a VPC network for both external resources to live in.
+1. The cloud administrator creates resource classes to be used when creating
+   `KubernetesCluster` and `MySQLInstance` resource claims against their cloud
+   of choice. These resource classes specify the necessary configuration to
+   ensure any managed resources backing the aforementioned claims can connect,
+   for example ensuring they're both configured to use the aforementioned VPC
+   network.
+1. The app operator creates their `KubernetesCluster` and `MySQLInstance`
+   resource claims, which either explicitly reference or default to the resource
+   classes created in step 2. Because their underlying managed resources are
+   appropriately configured, they can now communicate!
+
+The remainder of this proposal works through this scenario in each cloud
+provider, highlighting the changes necessary to enable connectivity.
+
+### Resource References
+
+It's typical for Kubernetes resources to refer to each other. The [Kubernetes
+API conventions] states:
+
+> References to loosely coupled sets of objects, such as pods overseen by a
+> replication controller, are usually best referred to using a label selector.
+> [...] References to specific objects, especially specific resource versions
+> and/or specific fields of those objects, are specified using the
+> ObjectReference type (or other types representing strict subsets of it).
+> [...] Object references should either be called fooName if referring to an
+> object of kind Foo by just the name (within the current namespace, if a
+> namespaced resource), or should be called fooRef, and should contain a subset
+> of the fields of the ObjectReference type.
+
+Crossplane currently uses these conventions to create references between
+resource claims and classes, resource claims and managed resources, etc. No
+pattern currently exists for modelling relationships between managed resources.
+
+Assume a `Network` managed resource named `kubernetesname` exists in namespace
+`crossplane-system`. Further assume this managed resource represents a GCP VPC
+network whose 'real' name in the GCP API is `externalname`. Finally, assume a
+`CloudSQLInstance` managed resource wants to specify this network as its
+`.spec.ipConfiguration.privateNetwork` (i.e. whitelisted VPC network). This
+could be implemented in one of two ways:
+
+1. The `CloudSQLInstance` simply requires `.spec.ipConfiguration.privateNetwork`
+   be provided as a `string` name specifying the real, external name of the
+   `Network`, i.e. `externalname`. It is up to the creator of the
```

Comment:
It does. Technically in the specific case of `CloudSQLInstance` it's a string like `/projects/mycoolproject/global/networks/externalname` rather than simply `externalname`.

I document that pattern when showing examples of `CloudSQLInstance` below, but had hoped to gloss over that detail in this section given that it's intended as an example to contrast simple string references vs Kubernetes object references, rather than documenting how `CloudSQLInstance` works.